polygeist/tools/cgeist/Test/Verification/addressspace.c

```diff
@@ -2,10 +2,8 @@
 
 // CHECK:  func.func @test() -> memref<?xi32, 4> attributes {llvm.linkage = #llvm.linkage<external>} {
 // CHECK:    [[CALL:%.*]] = call @foo() : () -> memref<?xi32, 1>
-// CHECK:    [[M2P:%.*]] = "polygeist.memref2pointer"([[CALL]]) : (memref<?xi32, 1>) -> !llvm.ptr<i32, 1>
-// CHECK:    [[ACAST:%.*]] = llvm.addrspacecast [[M2P]] : !llvm.ptr<i32, 1> to !llvm.ptr<i32, 4>
-// CHECK:    [[P2M:%.*]] = "polygeist.pointer2memref"([[ACAST]]) : (!llvm.ptr<i32, 4>) -> memref<?xi32, 4>
-// CHECK:    return [[P2M]] : memref<?xi32, 4>
+// CHECK:    [[ACAST:%.*]] = sycl.addrspacecast [[CALL]] : memref<?xi32, 1> to memref<?xi32, 4>
+// CHECK:    return [[ACAST]] : memref<?xi32, 4>
```

Comment:
This is a moot point because we are going to use the official memref operation to cast between memory spaces. That operation would be generated no matter that the element type of the memref is.  I really do not want to have a different code gen based on the -fsycl option because I think it makes our code diverge and inconsistent.

In this PR we can avoid loading the SYCL dialect but as soon as we can use the proper memref operation we will just load the memref dialect whether we compile with -fsycl or not.